analyzers/ruby/ruby.go

```diff
@@ -127,47 +127,85 @@ func (a *Analyzer) IsBuilt() (bool, error) {
 }
 
 func (a *Analyzer) Analyze() (graph.Deps, error) {
-	lockfilePath := filepath.Join(a.Module.Dir, "Gemfile.lock")
-	if a.Options.LockfilePath != "" {
-		lockfilePath = a.Options.LockfilePath
+	strategy := "list"
+
+	if a.Options.Strategy != "" {
+		strategy = a.Options.Strategy
 	}
 
-	switch a.Options.Strategy {
+	return a.analyzeHelper(strategy)
+}
+
+func (a *Analyzer) analyzeHelper(strategy string) (graph.Deps, error) {
+	lockfilePath := a.getLockfilePath()
+
+	switch strategy {
 	case "list":
-		gems, err := a.Bundler.List()
-		if err != nil {
-			return graph.Deps{}, err
-		}
-		imports, deps := FromGems(gems)
-		return graph.Deps{
-			Direct:     imports,
-			Transitive: deps,
-		}, nil
+		return a.bundlerListAnalyzerStrategy()
```

Comment:
We should not implicitly fall back if a strategy is set explicitly and manually by the user.